### PR TITLE
Add a pass to IR to extract procedure access operations (and a small feature update)

### PR DESCRIPTION
**Pull request by Romero027**
This PR adds a new pass (before the codegen) to analyze the procedure access pattern. We need this pass such that we can skip unnecessary RPC decodes and locks. (Also, for the rwlock to work). 

There is also a small feature update: I added a field call method name. It will be used when we have different RPC methods per edge. 

This will have merge conflicts with #59 .

### CODE REVIEW

File: compiler/element/frontend/printer.py
Comment (Kristoff-starling) [inline compiler/element/frontend/printer.py:1]: I don't think it appropriate to add the `WasmContext` type annotation, as for "Printer" visitor we don't need any context to generate IR summary, and what we pass to the `.accept()` method is a `None` value.

File: compiler/element/props/analyzer.py
Comment (Kristoff-starling) [inline compiler/element/props/analyzer.py:1]: Same as above.

**Review by Kristoff-starling — CHANGES_REQUESTED** [review]

**Comment by Kristoff-starling** [comment]
I have left some comments for minor issues. Please feel free to merge this PR first after addressing them, and I'll try to resolve the conflicts on my side after that :)

**Review by Kristoff-starling — APPROVED** [review]
LGTM!

**Review by livingshade — COMMENTED** [review]
LGTM!

**Pull request merged**